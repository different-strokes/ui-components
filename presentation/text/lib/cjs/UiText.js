'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// import { articleComponentPropTypes } from '@strokes/article-component-prop-types';

var UiText = function UiText(_ref) {
  var text = _ref.text,
      theme = _ref.theme;
  return _react2.default.createElement(
    'p',
    { className: theme.root },
    text
  );
};

UiText.propTypes = {
  // text: articleComponentPropTypes.text.isRequired,
  text: _propTypes2.default.string.isRequired,
  theme: _propTypes2.default.shape({
    root: _propTypes2.default.string
  }).isRequired
};

UiText.defaultProps = {
  theme: {}
};

exports.default = UiText;
//# sourceMappingURL=UiText.js.map