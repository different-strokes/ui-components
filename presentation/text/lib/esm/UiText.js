import React from 'react';
import PropTypes from 'prop-types';
// import { articleComponentPropTypes } from '@strokes/article-component-prop-types';

var UiText = function UiText(_ref) {
  var text = _ref.text;
  return React.createElement(
    'div',
    null,
    React.createElement(
      'p',
      null,
      'This is UiText. ',
      text
    )
  );
};

UiText.propTypes = {
  // text: articleComponentPropTypes.text.isRequired,
  text: PropTypes.string.isRequired
};

UiText.defaultProps = {};

export default UiText;
//# sourceMappingURL=UiText.js.map