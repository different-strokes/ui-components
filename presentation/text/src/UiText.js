import React from 'react';
import PropTypes from 'prop-types';
// import { articleComponentPropTypes } from '@strokes/article-component-prop-types';

const UiText = ({ text }) => (
  <div>
    <p>This is UiText. {text}</p>
  </div>
);

UiText.propTypes = {
  // text: articleComponentPropTypes.text.isRequired,
  text: PropTypes.string.isRequired,
};

UiText.defaultProps = {};

export default UiText;
