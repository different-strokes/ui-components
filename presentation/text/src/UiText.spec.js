import React from 'react';
import renderer from 'react-test-renderer';

import UiText from './UiText';

describe('UiText', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <UiText />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
