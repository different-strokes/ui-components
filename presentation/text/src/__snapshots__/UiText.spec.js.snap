// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiText renders correctly 1`] = `
<div>
  <p>
    This is UiText. 
  </p>
</div>
`;
